code/game/objects/items/weapons/medigun.dm

```diff
@@ -0,0 +1,364 @@
+			A.invisibility = INVISIBILITY_OBSERVER
+			user.visible_message("<span class='danger'> [user] busts the ghost of [A].</span>")
+			ghostbuster(A)
+			return
+		if(istype(A,/mob/living/simple_animal/shade))
+			var/mob/living/simple_animal/shade/S = A
+			for(var/i=0;i<3;i++)
+				new /obj/item/weapon/ectoplasm (S.loc)
+			user.visible_message("<span class='danger'> [user] busts \the [S].</span>")
+			ghostbuster(S)
+			S.ghostize()
+			qdel(S)
+			return
+
+		healtarget = A
+		wielder = user
+		processing_objects.Add(src)
+		playsound(user, fire_sound, 50, 1)
+
+		var/tracker_effect = "heal"
+		if(emagged)
+			tracker_effect += "red"
+		make_tracker_effects(get_turf(src), healtarget, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+		spawn(2)
+			make_tracker_effects(get_turf(src), healtarget, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+		spawn(4)
+			make_tracker_effects(get_turf(src), healtarget, 1, tracker_effect, 3, /obj/effect/tracker/heal)
+	else
+		playsound(user, empty_sound, 50, 1)
+
+/obj/item/medigun/attack_self(mob/user)
+	if((ubercharge >= MAX_UBERCHARGE) && healtarget)
+		ubercharge = 0
+		playsound(user, 'sound/weapons/medigun_ubercharge.ogg', 75, 1)
+
+		var/mob/living/ubertarget = healtarget
+		var/target_has_uberheart = 0
+		var/datum/organ/internal/heart/heartcheck = null
+
+		if(istype(ubertarget,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = ubertarget
+			heartcheck = H.internal_organs_by_name["heart"]
+			if("uber" in heartcheck.modifiers)
+				target_has_uberheart = 1
+
+		ubertarget.flags |= INVULNERABLE
+		animate(ubertarget, color = "#FF0000", time = 2, loop = -1)
+		animate(color = "#00FF00", time = 2)
+		animate(color = "#0000FF", time = 2)
+
+		if(emagged)
+			var/chance_of_death = 80//"Most hearts couldn't whistand this voltage...
+
+			if(target_has_uberheart)
+				chance_of_death = 20
+
+			if(prob(chance_of_death))//...But I'm fairly certain your heart-"...
+				to_chat(ubertarget,"<span class='danger'>THIS--THIS IS TOO MUCH POWER")
+				spawn(160)
+					if(ubertarget)
+						ubertarget.flags &= ~INVULNERABLE
+						ubertarget.gib()
+			else
+				to_chat(ubertarget,"<span class='danger'>YOU ARE BULLETPROOF")
+				spawn(80)
+					if(ubertarget)
+						animate(ubertarget)
+						ubertarget.color = null
+						ubertarget.flags &= ~INVULNERABLE
+						if(heartcheck)
+							heartcheck.damage += rand(5,60)
+		else
+			var/duration = 40
+
+			if(target_has_uberheart)
+				duration = 60
+
+			spawn(duration)
+				if(ubertarget)
+					animate(ubertarget)
+					ubertarget.color = null
+					ubertarget.flags &= ~INVULNERABLE
+					if(heartcheck && !target_has_uberheart)
+						heartcheck.damage += rand(5,30)
+
+
+/obj/item/medigun/emag_act(mob/user as mob)
+	if(!emagged)
+		to_chat(user, "<span class='warning'>You swipe the cryptographic sequencer through the circuits.</span>")
+		emagged = 1
+		medigunpack.emagged = 1
+		medigunpack.update_icon()
+		icon_state = "medigunred"
```

Comment:
The one you should have written on the medipack already to do exactly these sprite changes.